src/Features/RemotePaymentMethods/Init.php

```diff
@@ -0,0 +1,114 @@
+<?php
+/**
+ * Handles running payment method specs
+ */
+
+namespace Automattic\WooCommerce\Admin\Features\RemotePaymentMethods;
+
+defined( 'ABSPATH' ) || exit;
+
+use Automattic\WooCommerce\Admin\RemoteInboxNotifications\SpecRunner;
+
+/**
+ * Remote Payment Methods engine.
+ * This goes through the specs and gets eligible payment methods.
+ */
+class Init {
+	const SPECS_TRANSIENT_NAME = 'woocommerce_admin_remote_payment_methods_specs';
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		add_action( 'change_locale', array( __CLASS__, 'delete_specs_transient' ) );
+	}
+
+	/**
+	 * Go through the specs and run them.
+	 */
+	public static function get_methods() {
+		$methods = array();
+		$specs   = self::get_specs();
+
+		foreach ( $specs as $spec ) {
+			$method    = EvaluateMethod::evaluate( $spec );
+			$methods[] = $method;
+		}
+
+		return $methods;
+	}
+
+	/**
+	 * Delete the specs transient.
+	 */
+	public static function delete_specs_transient() {
+		delete_transient( self::SPECS_TRANSIENT_NAME );
+	}
+
+	/**
+	 * Get specs or fetch remotely if they don't exist.
+	 */
+	public static function get_specs() {
+		$specs = get_transient( self::SPECS_TRANSIENT_NAME );
+
+		// Fetch specs if they don't yet exist.
+		if ( false === $specs || ! is_array( $specs ) || 0 === count( $specs ) ) {
+			// We are running too early, need to poll data sources first.
+			$specs = DataSourcePoller::read_specs_from_data_sources();
+			$specs = self::localize( $specs );
+			set_transient( self::SPECS_TRANSIENT_NAME, $specs, 7 * DAY_IN_SECONDS );
```

Comment:
Would it be possible to set this transient to something super small for dev environments? Maybe we could go off `WP_DEBUG`?

Reply:
Good question.  As part of the testing in #6867 I added a remote payments tester plugin that filters the transient so the data is forced to fetch each time. 

https://github.com/joshuatf/remote-payments-tester/blob/62af802d3ab588ba22a0663958de7ac886fa0735/remote-payments-tester.php#L29